### PR TITLE
Enabling bounded dequeue stash and receive timeout test

### PR DESCRIPTION
**Pull request by rogeralsing**
Enabling some tests that previously have been set to Skip due to broken or not implemented code.

### CODE REVIEW

File: src/core/Akka.Tests/Actor/StashMailboxSpec.cs [hunk 1]
```diff
@@ -32,13 +32,12 @@ public void When_creating_actor_marked_with_WithUnboundedStash_a_mailbox_which_s
...
 
-        [Fact(Skip = "We do not have a BoundedDequeBasedMailbox yet")] //TODO: Remove Skip when we have a BoundedDequeBasedMailbox
+        [Fact]
         public void When_creating_actor_marked_with_WithBoundedStash_a_mailbox_which_supports_unbounded_stash_is_created()
         {
             var actorRef = ActorOf<BoundedStashActor>();
             var intRef = (LocalActorRef)actorRef;
-            //intRef.Cell.Mailbox.GetType().ShouldBe(typeof(BoundedDequeBasedMailbox));
-            throw new Exception("Incomplete. Remove the comment on the line above this, and remove this line, when we have BoundedDequeBasedMailbox");
+            intRef.Cell.Mailbox.GetType().ShouldBe(typeof(BoundedDequeBasedMailbox));
```
Comment (rogeralsing) [inline src/core/Akka.Tests/Actor/StashMailboxSpec.cs]: The BoundedDequeBasedMailbox have been ported for quite some time ago.
Test now passes.

File: src/core/Akka/Actor/ActorCell.ReceiveTimeout.cs [hunk 2]
```diff
@@ -19,12 +19,20 @@ public void SetReceiveTimeout(TimeSpan? timeout=null)
...
         }
 
+        public TimeSpan? ReceiveTimeout
+        {
+            get
+            {
+                return _receiveTimeoutDuration;
+            }
+        }
+
```
Comment (rogeralsing) [inline src/core/Akka/Actor/ActorCell.ReceiveTimeout.cs]: ReceiveTimeout getter is available in the Scala code base (together with SetReceiveTimeout) so we are inline with them.
This property is needed in the `TestActorRef_must_set_ReceiveTimeout_to_None` test (also ported from Scala)

File: src/core/Akka.TestKit.Tests/TestActorRefTests/TestActorRefSpec.cs [hunk 3]
```diff
@@ -143,11 +143,11 @@ public void TestActorRef_must_allow_access_to_internals()
             Assert.Equal("Hejsan!", actor.ReceivedString);
         }
 
-        [Fact(Skip = "Context.ReceiveTimeout is not implemented. Test cannot run")]
+        [Fact]
         public void TestActorRef_must_set_ReceiveTimeout_to_None()
         {
             var a = new TestActorRef<WorkerActor>(Sys, Props.Create<WorkerActor>());
-            //TODO: When Context.ReceiveTimeout is implemented: Assert.Equal(((IInternalActor)a.UnderlyingActor).ActorContext.ReceiveTimeout, not sure what value to put here: null or Timeout.InfiniteTimeSpan);
+            ((IInternalActor)a.UnderlyingActor).ActorContext.ReceiveTimeout.ShouldBe(null);
```
Comment (rogeralsing) [inline src/core/Akka.TestKit.Tests/TestActorRefTests/TestActorRefSpec.cs]: This test is from the Scala tests.

File: src/core/Akka/Actor/IActorContext.cs [hunk 4]
```diff
@@ -135,6 +135,11 @@ public interface IActorContext : IActorRefFactory, ICanWatch
         /// <param name="timeout">The timeout. Pass in <c>null</c> to switch off this feature.</param>
         void SetReceiveTimeout(TimeSpan? timeout);
 
+        /// <summary>
+        /// Gets the inactivity deadline timeout set using <see cref="SetReceiveTimeout"/>.
+        /// </summary>
+        TimeSpan? ReceiveTimeout { get; }
```
Comment (rogeralsing) [inline src/core/Akka/Actor/IActorContext.cs]: This does not version well, Adding members to interfaces.
But..
It is a member in Scala and thus belongs here. and there is currently no way of checking if receive timeout has been set to a value.
It is also very unlikely that there are custom implementations of the interface out there that does not inherit `ActorCell`.

Thoughts on how to handle this? there are probably more places where we haven't yet fully ported the Scala code and need to add more members to support new features.

**Comment by Horusiath** [comment]
Looks good for me.

**Pull request merged**